File: judge-api/src/main/kotlin/io/leonfoliveira/judge/api/controller/dto/response/ContestResponseDTO.kt
package io.leonfoliveira.judge.api.controller.dto.response

import io.leonfoliveira.judge.core.domain.entity.Contest
import java.time.LocalDateTime

class ContestResponseDTO(
    val id: Int,
    val title: String,
    val startAt: LocalDateTime,
    val endAt: LocalDateTime,
)

fun Contest.toResponseDTO(): ContestResponseDTO {
    return ContestResponseDTO(
        id = this.id,
        title = this.title,
        startAt = this.startAt,
        endAt = this.endAt,
    )
}

File: judge-web/src/core/repository/dto/response/ContestResponseDTO.ts
import { Language } from "@/core/domain/enumerate/Language";
import { MemberResponseDTO } from "@/core/repository/dto/response/MemberResponseDTO";
import { ProblemResponseDTO } from "@/core/repository/dto/response/ProblemResponseDTO";

export type ContestResponseDTO = {
  id: number;
  title: string;
  languages: Language[];
  startAt: string;
  endAt: string;
  members: MemberResponseDTO[];
  problems: ProblemResponseDTO[];
};
